### DISCUSSION TITLE
Remove "i" Info Button

### DISCUSSION THREAD

**Discussion by shaykbake** [DirectoryLister/DirectoryLister#1160] [Ideas & Features]
Hello. Perhaps this can already be done with the existing configuration, but I couldn't figure it out.

When I disable hashing (setting MAX_HASH_SIZE to either false, null or 0), I still have the "i" info button for each file when hovering, but this button just opens a loading animation into the middle of the page since hashing is basically disabled.

It would be nice if we could remove this button altogether when generating hashes for the files is disabled, since this appears to be the only purpose of the button.

Thanks!

**Comment by ccizm** [comment]
I tried modifying the configuration but it didn't work, Currently this effect can be achieved by deleting the code in the file.



open [/app/views/components/file.twig](https://github.com/DirectoryLister/DirectoryLister/blob/master/app/views/components/file.twig?plain=1#L18-L26)



Delete this

```
        {% if file.isFile %}
            <div class="ml-2">
                <button title="{{ translate('file.info') }}" @click.prevent="$dispatch('show-file-info', { file: '{{ url(file.getPathname) | escape('js') }}' })"
                    class="justify-center items-center rounded-full w-8 h-8 -m-1 md:invisible hover:bg-gray-300 hover:shadow dark:hover:bg-purple-900 group-hover:visible"
                >
                    <i class="fas fa-info-circle"></i>
                </button>
            </div>
        {% endif %}
```

**Reply by shaykbake** [reply]
You helped me accomplish exactly what I wanted. Instead of just removing that, this edit to the condition check appears to only show the info button for files where the `max_hash_size` config variable set in the `.env` file is large enough in comparison to the file size for each file listed in the directory:

REPLACE
`{% if file.isFile %}`
WITH
`{% if file.isFile and file.getSize <= config('max_hash_size') %}`

You have to delete the `cache` directory in the DirectoryLister's main directory to see the changes. I unfortunately spent more time than I should have trying to get this to work because I didn't realize it wasn't actually showing me those changes...

I tested this with two files of different sizes, and this does allow DL to still provide the info button and hashes for files within size limits but hides the info button for those that exceed it.

Thanks for the help!
